src/output/help.rs

```diff
@@ -1117,6 +1121,7 @@ const TAB_WIDTH: usize = 4;
 pub(crate) enum HelpWriter<'writer> {
     Normal(&'writer mut dyn Write),
     Buffer(&'writer mut Colorizer),
+    String(&'writer mut String),
```

Comment:
Why is a new `HelpWriter` needed?

Reply:
This was to simplify writing to a String, but from the points you have made it shouldn't be necessary.